examples/apps/simple_wallet/workload/simple_wallet_execute_io.cpp

```diff
@@ -0,0 +1,187 @@
+#include "tcf_error.h"
+#include "enclave_utils.h"
+#include <parson.h>
+
+#include "simple_wallet_execute_io.h"
+#include "trusted_iohandler.h"
+#include "crypto_iohandler.h"
+
+#define LEDGER_FILE "/tmp/account_ledger.json"
+
+// Placeholder to store all ledger entries. Useful when updating
+// given account in the ledger.
+std::string g_ledgerDetails;
+
+// Create Wallet ledger file to store details of transactions in encrypted format
+void CreateWalletLedger() {
+    uint32_t status = ExecuteFileWrite("");
+    if (status == 0) {
+        Log(TCF_LOG_INFO, "Successfully created wallet ledger file %s",
+            LEDGER_FILE);
+    } else {
+        Log(TCF_LOG_INFO, "Failed to create wallet ledger file %s",
+            LEDGER_FILE);
+    }
+}
+
+// Get wallet balance for given account from json serialized ledger
+std::string GetWalletBalanceFromLedger(std::string ledgerData,
+    std::string accountName) {
+    // Parse account_ledger.json data
+    JsonValue parsed(json_parse_string(ledgerData.c_str()));
+    tcf::error::ThrowIfNull(parsed.value,
+        "failed to parse the work order request, badly formed JSON");
+
+    JSON_Object* ledger_object = json_value_get_object(parsed);
+    tcf::error::ThrowIfNull(ledger_object, "Missing JSON object in ledger");
+    Log(TCF_LOG_INFO, "Getting balance for the account from ledger object: %s",
+        accountName);
+    std::string balance = GetJsonStr(ledger_object, accountName.c_str(),
+        nullptr);
+    Log(TCF_LOG_INFO, "GetWalletBalanceFromLedger::Got balance for account %s",
+        accountName);
+    return balance;
+}
+
+std::string UpdateLedger(std::string ledgerData, std::string accountName,
+    std::string amount) {
+    JSON_Value* ledgerVal;
+    if (ledgerData.empty()) {
+        Log(TCF_LOG_INFO, "Creating first wallet in ledger");
+        ledgerVal = json_value_init_object();
+    } else {
+        // Parse account_ledger.json data
+        ledgerVal = json_parse_string(ledgerData.c_str());
+        tcf::error::ThrowIfNull(
+        ledgerVal, "failed to parse wallet ledger. Poorly formatted json");
+    }
+
+    JSON_Object* ledgerObj = json_value_get_object(ledgerVal);
+    tcf::error::ThrowIfNull(ledgerObj, "Missing JSON object in ledger");
+
+    JsonSetStr(ledgerObj, accountName.c_str(), amount.c_str(),
+        "failed to serialize wallet details");
+    // Serialize the resulting json
+    size_t serializedSize = json_serialization_size(ledgerVal);
+    ByteArray serialized_response;
+    serialized_response.resize(serializedSize);
+
+    JSON_Status jret = json_serialize_to_buffer(ledgerVal,
+        reinterpret_cast<char*>(&serialized_response[0]),
+        serialized_response.size());
+    tcf::error::ThrowIf<tcf::error::RuntimeError>(
+        jret != JSONSuccess, "Wallet serialization failed");
+
+    return std::string(serialized_response.begin(), serialized_response.end());
+
+}
+
+// Read ledger file, decrypt content and store the data in ledgerData
```

Comment:
Couldn't find any decrypt operation in this function

Reply:
right, will update the comment.